codalab/lib/bundle_cli.py

```diff
@@ -1939,8 +1939,21 @@ def do_macro_command(self, args):
         if not getattr(args, metadata_util.metadata_key_to_argument('name')):
             setattr(args, metadata_util.metadata_key_to_argument('name'), 'new')
         # Reduce to the mimic case
-        args.bundles = [args.macro_name + '-in' + str(i+1) for i in range(len(args.bundles))] + \
-                       [args.macro_name + '-out'] + args.bundles
+        named_inputs = [bundle for bundle in args.bundles if ':' in bundle]
+        named_input_names = [bundle.split(':',1)[0] for bundle in named_inputs]
+
+        named_user_inputs = [bundle.split(':',1)[1] for bundle in named_inputs]
+        named_macro_inputs = [args.macro_name + '-in-' + named_in for named_in in named_input_names]
+
+        numbered_user_inputs = [bundle for bundle in args.bundles if ':' not in bundle]
```

Comment:
Would be better to have the logic for checking `:` in bundle deduped / more localized - first partition the args (`itertools.partition`), and then process each one.

Reply:
`itertools.partition` isn't a python builtin but a recipe they recommend using. I didn't want to define an inline partition function just for this, so I also just encompassed this logic in the bundle-by-bundle processing for loop.